tests/test_custom_nonce.py

```diff
@@ -0,0 +1,62 @@
+import os
+import json
+import pytest
+import secp256k1
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+DATA = os.path.join(HERE, 'data')
+
+from cffi import FFI
+
+ffi = FFI()
+ffi.cdef('static int nonce_function_rand(unsigned char *nonce32,'
+         'const unsigned char *msg32,const unsigned char *key32,'
+         'const unsigned char *algo16,void *data,unsigned int attempt);')
+
+#The most elementary conceivable nonce function, acting
+#as a passthrough: user provides random data which must be
+#a valid scalar nonce. This is not ideal, 
+#since libsecp256k1 expects the nonce output from
+#this function to result in a valid sig (s!=0), and will
+#increment the counter ("attempt") and try again if it fails;
+#since we don't increment the counter here, that will not succeed.
+#Of course the likelihood of such an error is infinitesimal.
+#TLDR this is not intended to be used in real life; use
+#deterministic signatures.
```

Comment:
As briefly mentioned here, using a custom nonce function is certainly not something you'd normally want to do; the particular use case we had related to stealth addresses. The problem was that the default operation of the library ecdsa_sign does not return the value of the nonce, and this is needed in this case. A brief discussion with Greg Maxwell today indicated that there might be a possibility of having the nonce value returned in the extra data field in future in the upstream library; this would allow one to use the nonce value without using a non-default (non-rfc6979) nonce. There *may* be other situations where a custom nonce function is required for some reason, so one way or another this patch (which is just exposing what the underlying library offers) might conceivably be useful for someone, even given the caveats.